src/app/util/CHIPDeviceCallbacksMgr.cpp

```diff
@@ -64,8 +64,22 @@ CHIP_ERROR CHIPDeviceCallbacksMgr::AddResponseCallback(NodeId nodeId, uint8_t se
     VerifyOrReturnError(onFailureCallback != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
     ResponseCallbackInfo info = { nodeId, sequenceNumber };
-    memcpy(&onSuccessCallback->mInfoPtr, &info, sizeof(info));
-    memcpy(&onFailureCallback->mInfoPtr, &info, sizeof(info));
+
+    onSuccessCallback->mInfoPtr = CHIPPlatformMemoryAlloc(sizeof(info));
+    if (onSuccessCallback->mInfoPtr == nullptr)
+    {
+        return CHIP_ERROR_NO_MEMORY;
+    }
+
+    onFailureCallback->mInfoPtr = CHIPPlatformMemoryAlloc(sizeof(info));
+    if (onFailureCallback->mInfoPtr == nullptr)
+    {
+        CHIPPlatformMemoryFree(onSuccessCallback->mInfoPtr);
+        return CHIP_ERROR_NO_MEMORY;
+    }
+
+    memcpy(onSuccessCallback->mInfoPtr, &info, sizeof(info));
+    memcpy(onFailureCallback->mInfoPtr, &info, sizeof(info));
 
     // If some callbacks have already been registered for the same ResponseCallbackInfo, it usually means that the response
```

Comment:
The CancelCallback should be moved up to before any alloc of new callbacks, so that they release prior memory